### ISSUE TITLE
handle errors

### ISSUE THREAD

**Issue by medchbani** [JocaPC/express4-tedious#2]
Hi ,
Please is there any way to handle errors when executing a query, example:
```
req.query("select * from t_users for json path")
        .into(res, '[]');
```
Regards.

**Comment by JocaPC** [comment]
Hi,
There is a `fail` method that you can use:

    req.query("select * from t_users for json path")
          .fail(function(ex) { /* do something with exception*/    } )
          .into(res, '[]');

You also have `done()` method that will be called always.

**Comment by medchbani** [comment]
Thanks for your reply,
i still have one issue : 
i can see error message in console but how to return error message as response.

```
 req.query("select * from products for json path").fail(function(ex) { 
	  		console.log(ex.message); 
	   } )
        .into(res,'[]');
```

**Comment by JocaPC** [comment]
Could you use res object in the error handler (e.g. res.write)?

**Comment by medchbani** [comment]
impossible to use res.write because its says the res its already sent.

`Caught exception: Error: Can't set headers after they are sent.`

**Comment by JocaPC** [comment]
I had to redesign whole module to fix this. I have tried it and it is fixed in version 0.2.
